bin.src/inspect_job.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+#
+# This file is part of ap_verify.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Print the contents of a persisted Job object.
+
+This script takes as arguments one or more Job .json files, and prints the
+top-level metadata and a summary of any measurements. It does not print
+metrics that don't have measurements (there are far too many) or
+specifications (which are not helpful for testing measurement code).
+"""
+
+import sys
+import json
+
+from lsst.verify import Job
+
+
+def is_measurement_metadata(key, metrics):
+    """Test whether a job-level metadata key is really measurement metadata.
+
+    Parameters
+    ----------
+    key : `str`
+        The key to test.
+    metrics : iterable of `lsst.verify.Name` or of `str`
+        The metrics recorded in the job.
+
+    Returns
+    -------
+    result : `bool`
+        `True` if ``key`` represents measurement metadata, `False` otherwise.
+    """
+    for metric in metrics:
+        if str(metric) in key:
+            return True
+    return False
+
+
+def simplify_key(key, prefix):
+    """Remove a prefix from a key, if it's present.
+
+    Parameters
+    ----------
+    key : `str`
+        The key to simplify.
+    prefix : `str`
+        The prefix to remove from ``key``.
+
+    Returns
+    -------
+    simplifiedKey : `str`
+        ``key`` with any initial ``prefix`` removed
+    """
+    if key.startswith(prefix):
+        return key.replace(prefix, "", 1)
+    else:
+        return key
+
+
+def inspect_job(filename):
+    """Present the measurements in a Job file.
+
+    The measurements and any metadata are printed to standard output.
+
+    Parameters
+    ----------
+    filename : `str`
+        The Job file to open. Must be in JSON format.
+    """
+    with open(filename) as f:
+        job = Job.deserialize(**json.load(f))
+
+    maxMetricLength = max([len(str(metric)) for metric in job.measurements])
```

Comment:
A comment that this is used to format the string so everything is the same "=" level? (at least, I think that's what it's for)